src/skill.cpp

```diff
@@ -205,27 +205,66 @@ bool Skill::is_contextual_skill() const
 
 void SkillLevel::train( int amount, bool skip_scaling )
 {
-    // Working off rust to regain levels goes twice as fast as reaching levels in the first place
+    int catchup_amount = amount;
+    int highest_level_exp = _highestLevel * _highestLevel * 10000;
+
+    // Working off rust to regain levels goes twice as fast as reaching levels in the first place, and 1.5x as fast to regain your last place in the level.
```

Comment:
Perhaps the effect of SKILL_RUST set to "Int" should be the "amount" being multiplied by more with higher Intelligence?

Reply:
That would be worth considering yes, in fact all the things that currently work to slow skill rust should instead give a boost to regaining skill after rust.

Reply:
Regaining skill (increasing that x2) and/or boosting your theoretical knowledge (increasing that x1.1). I'm thinking intelligence should favor the latter, while memory boosters should favor the former.

Reply:
That is a reasonable idea. I'm not sure I'll get to it by this PR, we'll see, but if not I'll suggest it in follow up